src/Engines/nativeEngine.ts

```diff
@@ -884,6 +935,59 @@ export class NativeEngine extends Engine {
 
         // Shader processor
         this._shaderProcessor = new NativeShaderProcessor();
+
+        if (!this.originBottomLeft) {
+            // prepare YFlip shader and vertex stream
+            Effect.ShadersStore["fullscreenFlipVertexShader"] = `
+                attribute vec2 position;
+                attribute vec2 uv;
+                varying vec2 vUV;
+                void main(void) {
+                    gl_Position = vec4(position, 0.5, 1.0);
+                    vUV = position * 0.5 + 0.5;
+                    vUV.y = 1.0 - vUV.y;
+                }`;
+            Effect.ShadersStore["fullscreenFlipFragmentShader"] = `
+                varying vec2 vUV;
+                uniform sampler2D backBuffer;
+                void main(void)
+                {
+                    vec3 color = texture2D(backBuffer, vUV).xyz;
+                    gl_FragColor = vec4(color, 1.0);
+                }`;
+
+            this._YFlip = {
+                vertexBuffers: { [VertexBuffer.PositionKind]: new VertexBuffer(this, [3, -1, -1, -1, -1, 3], VertexBuffer.PositionKind, false, false, 2) },

```

Comment:
Is it faster to draw a single big triangle that goes outside the limit of the windows than two triangles that fit exactly in?

Reply:
Technically, yes. GPU computes 2x2 Tiles. With 2 triangles, the 2x2 tiles on the screen diagonal would be computed twice. In practice, it won't make a difference. It's just using a bit less memory, less code,...